### DISCUSSION TITLE
Serializing and deserializing data between Llama Index and Llama Index TS

### DISCUSSION THREAD

**Discussion by preemoDez** [run-llama/LlamaIndexTS#526] [Q&A] [answered]
Hello,

I wonder if it's possible to do the following things:

1.
- get the documents from `SimpleDirectoryReader` and convert them to JSON string;
- send it over the network to the Python service, deserialize them, and get nodes from documents
2.
- serialize the collection of nodes to JSON string in Python
- send it over the network to the service in TS, and deserialize them by calling `jsonToNode`.

Is there any contract or guarantee that it won't break in the future?

**Comment by marcusschiesser** [comment, marked as answer]
@preemoDez, in the current state, we don't have a guaranteed contract. Compatibility depends on the storage used and the library's version. Data saved using the Mongo DB vector store is currently interchangeable (class `MongoDBAtlasVectorSearch`).
It's definitely something we need to work on. Thanks @preemoDez !

**Comment by preemoDez** [comment]
Thanks for the info!
